### ISSUE TITLE
Support for iPad?

### ISSUE THREAD

**Issue by felixschulze** [domesticcatsoftware/DCIntrospect#26]
Has anyone got this working for iPad? I'm getting only a black screen with DCIntrospect on it.

**Comment by gradha** [comment]
I've been using it with iPad apps all the time. You can open the demo provided in the project and change the device family to ipad. The demo UI will look bad on the ipad, but everything should be functional.

**Comment by vpdn** [comment]
I am also just getting a black screen (http://screencast.com/t/UXpxN3fSdB) on the iPad 4.3 emulator.

It seems the issue has to do with the calculation of the statusbar location. The black screen you're seeing is actually the status bar overlay, drawn with the dimension (w:1024, h:1024). As a quick check, try commenting out the code in  "updateStatusBar" in the DCIntrospect class.

**Comment by gradha** [comment]
I think I found the culprit. Do your applications launch into portrait or landscape mode? DCIntrospect works fine when the application supports portrait orientations, but seems to fail when you restrict the application to landscape. I tried updating DCStatusBarOverlay::updateBarFrame replacing all instances of self.frame.size.\* with the constant 20, and then the bar is resized correctly.

However, possibly due to the _unexpected_ launch orientation, the transforms don't work as expected and the overlay status is left floating in the middle of the screen... more precisely, at offset (748, 505) as measured by DCIntrospect itself.

**Comment by vpdn** [comment]
>  DCIntrospect works fine when the application supports portrait orientations, but seems to fail when you restrict the application to landscape.

Yes, and the problem can be reproduced with the DCIntospect Demo as well. Simply set shouldAutorotateToInterfaceOrientation to UIInterfaceOrientationIsLandscape(interfaceOrientation).

**Comment by gradha** [comment]
I've tried to add a static variable to remember the first orientation the method goes through and use that to counter the expected rotation. See https://github.com/gradha/DCIntrospect/commit/bae3260b4477c731f8c0029174b2f2cdda788272
Now the black bar seems to be positioned and rotated correctly but I'm missing the labels.

**Comment by gradha** [comment]
Ok, ignore that one, I was looking at the wrong place, this should work better: https://github.com/gradha/DCIntrospect/commit/548afd9cbec17f210b902d4c592d1419da58a7c1

The help still comes up rotated incorrectly, but the bar should not be a problem any more. See if it works for you.

**Comment by vpdn** [comment]
Looks good. This does fix the status bar orientation issue for me.

**Comment by xiakehc** [comment]
need restart Project and setting devices is iPad,
